internal/controller/pipelinerollout_controller.go

```diff
@@ -687,22 +704,22 @@ func (r *PipelineRolloutReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return nil
 }
 
-// remove 'lifecycle.desiredPhase' key/value pair from Pipeline spec
+// remove 'lifecycle.desiredPhase' key/value pair from spec
 // also remove 'lifecycle' if it's an empty map
-func pipelineWithoutDesiredPhase(obj *kubernetes.GenericObject) (map[string]interface{}, error) {
-	var pipelineAsMap map[string]any
-	if err := json.Unmarshal(obj.Spec.Raw, &pipelineAsMap); err != nil {
+func withoutDesiredPhase(obj *kubernetes.GenericObject) (map[string]interface{}, error) {
+	var specAsMap map[string]any
+	if err := json.Unmarshal(obj.Spec.Raw, &specAsMap); err != nil {
 		return nil, err
 	}
 	// remove "lifecycle.desiredPhase"
 	comparisonExcludedPaths := []string{"lifecycle.desiredPhase"}
-	util.RemovePaths(pipelineAsMap, comparisonExcludedPaths, ".")
+	util.RemovePaths(specAsMap, comparisonExcludedPaths, ".")
 	// if "lifecycle" is there and empty, remove it
-	lifecycleMap, found := pipelineAsMap["lifecycle"].(map[string]interface{})
+	lifecycleMap, found := specAsMap["lifecycle"].(map[string]interface{})
 	if found && len(lifecycleMap) == 0 {
-		util.RemovePaths(pipelineAsMap, []string{"lifecycle"}, ".")
+		util.RemovePaths(specAsMap, []string{"lifecycle"}, ".")
 	}
-	return pipelineAsMap, nil
+	return specAsMap, nil
```

Comment:
I am using the same function in `MonoVertexRolloutReconciler` (in my other not-yet-merged branch that I forked this from) which is why I renamed this. Ideally, maybe it should be moved to another file.